src/dtls13.c

```diff
@@ -0,0 +1,2432 @@
+
+        /* the other peer may have retransmitted because an ACK for a flight
+           that needs explicit ACK was lost.*/
+        if (ssl->dtls13Rtx.seenRecords != NULL)
+            ssl->dtls13Rtx.sendAcks = ssl->options.sendMoreAcks;
+    }
+
+    if (ssl->keys.dtls_peer_handshake_number ==
+            ssl->keys.dtls_expected_peer_handshake_number &&
+        ssl->options.handShakeDone && hs == certificate_request) {
+
+        /* the current record, containing a post-handshake certificate request,
+           is implicitly acknowledged by the
+           certificate/certificate_verify/finished flight we are about to
+           send. Please note that if the certificate request came out-of-order
+           and we didn't send an ACK (sendMoreAcks == 0 and the missing
+           packet(s) arrive before that fast timeout expired), then we will send
+           both the ACK and the flight. While unnecessary this it's harmless, it
+           should be rare and simplifies the code. Otherwise, it would be
+           necessary to track which record number contained a CertificateRequest
+           with a particular context id */
+        Dtls13RtxRemoveCurAck(ssl);
+    }
+
+    if (ssl->options.sendMoreAcks && Dtls13DetectDisruption(ssl, fragOffset)) {
+        WOLFSSL_MSG("Disruption detected");
+        ssl->dtls13Rtx.sendAcks = 1;
+    }
+
+    return 0;
+}
+
+void Dtls13FreeFsmResources(WOLFSSL *ssl)
+{
+    Dtls13RtxFlushAcks(ssl);
+    Dtls13RtxFlushBuffered(ssl, 0);
+}
+
+static int Dtls13SendOneFragmentRtx(WOLFSSL *ssl,
+    enum HandShakeType handshakeType, word16 outputSize, byte *message,
+    word32 length, int hashOutput)
+{
+    Dtls13RtxRecord *rtxRecord;
+    word16 recordHeaderLength;
+    byte isProtected;
+    int ret;
+
+    isProtected = Dtls13TypeIsEncrypted(handshakeType);
+    recordHeaderLength = Dtls13GetRlHeaderLength(isProtected);
+
+    rtxRecord = Dtls13RtxNewRecord(ssl, message + recordHeaderLength,
+        length - recordHeaderLength, handshakeType,
+        ssl->dtls13EncryptEpoch->nextSeqNumber);
+
+    if (rtxRecord == NULL)
+        return MEMORY_E;
+
+    ret = Dtls13SendFragment(ssl, message, outputSize, length, handshakeType,
+        hashOutput, Dtls13SendNow(ssl, handshakeType));
+
+    if (ret == 0 || ret == WANT_WRITE)
+        Dtls13RtxAddRecord(&ssl->dtls13Rtx, rtxRecord);
+    else
+        Dtls13FreeRtxBufferRecord(ssl, rtxRecord);
+
+    return ret;
+}
+
+static int Dtls13SendFragmentedInternal(WOLFSSL *ssl)
+{
+    int fragLength, rlHeaderLength;
+    int remainingSize, maxFragment;
+    int recordLength;
+    byte isEncrypted;
+    byte *output;
+    int ret;
+
+    isEncrypted = Dtls13TypeIsEncrypted(ssl->dtls13FragHandshakeType);
+    rlHeaderLength = Dtls13GetRlHeaderLength(isEncrypted);
+    maxFragment = wolfSSL_GetMaxFragSize(ssl, MAX_RECORD_SIZE);
+
+    remainingSize = ssl->dtls13MessageLength - ssl->dtls13FragOffset;
+
+    while(remainingSize > 0) {
```

Comment:
space after while

Reply:
missed that! I'm gonna fix it in the next review cycle